src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -0,0 +1,84 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import com.google.sps.data.HuntItem;
+import com.google.sps.data.Riddle;
+import com.google.sps.data.ScavengerHunt;
+import java.io.IOException;
+import java.util.ArrayList;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/go-data")
+public class DataServlet extends HttpServlet {
+  private static final String JSON_TYPE = "application/json";
+
+  private ScavengerHunt hunt;
+  private int index = -1;
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String indexStr = request.getParameter("new-index");
+    index = Integer.parseInt(indexStr);
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    buildScavengerHunt();
+
+    response.setContentType(JSON_TYPE);
+    Gson gson = new Gson();
+    String json = gson.toJson(hunt);
+    response.getWriter().println(json);
+  }
+
+  public void buildScavengerHunt() {
+    // Constructing the first HuntItem.
+    ArrayList<String> firstHints = new ArrayList<String>();
+    firstHints.add("I am at the periphery of SF");
+    firstHints.add("I am golden in color");
+    Riddle firstRiddle = new Riddle("I was constructed in 1933", firstHints);
+    HuntItem firstHunt =
+        new HuntItem(
+            "Golden Gate Bridge", "San Francisco", "A famous bridge in San Francisco", firstRiddle);
+
+    // Constructing the second HuntItem.
+    ArrayList<String> secondHints = new ArrayList<String>();
+    secondHints.add("I am very tall");
+    secondHints.add("I am a popular tourist destination");
+    Riddle secondRiddle = new Riddle("A famous tower in Paris", secondHints);
+    HuntItem secondHunt =
+        new HuntItem("Eiffel Tower", "Paris", "A famous tower in Paris", secondRiddle);
+
+    // Constructing the third HuntItem.
+    ArrayList<String> thirdHints = new ArrayList<String>();
+    thirdHints.add("I am a statue");
+    thirdHints.add("I am very tall");
+    Riddle thirdRiddle = new Riddle("I am an item often associated with America", thirdHints);
+    HuntItem thirdHunt =
+        new HuntItem("Statue of Liberty", "NYC", "A statue in New York City", thirdRiddle);
```

Comment:
Since you're trying to build these HuntItem  objects, I think a builder pattern would be great here:
https://dzone.com/articles/design-patterns-the-builder-pattern

What you would do is in the HuntItem class, create a Builder sub-class with a couple methods like withName, withRiddles, etc.

Your code would be much more readable:

` HuntItem item = new HuntItem.Builder().withName("Statue of Liberty").withCity("NYC").build();`

Same for the Riddle class.  You could have something that looked like this:
`new Riddle.Builder().withPuzzle("I am an item often associated with America").withRiddle("I am very tall").withRiddle("I am a statue").build();`

This is all dummy data for now, but eventually you would need to construct these objects with data pulled from the datastore.  I think if your classes used the builder pattern it might make constructing these objects from the datastore objects easier.

Let me know if you have any questions!

Reply:
thanks for letting me know! The code looks a lot cleaner now.